packages/lightspark-sdk/src/tests/integration/general-regtest.test.ts

```diff
@@ -0,0 +1,461 @@
+    })
+
+    // TODO: THIS ACTION CAN BE CREATED ONLY IN MAINNET
+    // test('Should deposit funds to wallet with a defined amount of sats', async () => {
+    //     const fundingResult = await lightsparkClient.requestWithdrawal(getRegtestNodeId(), PAY_AMOUNT, '', WithdrawalMode.WALLET_THEN_CHANNELS)
+    //     const transaction = await lightsparkClient.waitForTransactionComplete(fundingResult.id, TRANSACTION_WAIT_TIME)
+    //     expect(transaction.status).toBe(TransactionStatus.SUCCESS)
+    // }, TRANSACTION_WAIT_TIME)
+
+    test('Should open just-in-time channel from inbound payment', async () => {
+        const regtestNodeId = getRegtestNodeId()
+        const testInvoice = await lightsparkClient.createInvoice(
+            regtestNodeId,
+            PAY_AMOUNT,
+            'hi there!'
+        )
+
+        if(!testInvoice) {
+            throw new TypeError('Test invoice wasn\'t created')
+        }
+
+        const payment = await lightsparkClient.createTestModePayment(
+            regtestNodeId,
+            testInvoice
+        )
+        if(!payment) {
+            throw new TypeError('Test mode payment wasn\'t created')
+        }
+        const transaction = await lightsparkClient.waitForTransactionComplete(
+            payment.id,
+            TRANSACTION_WAIT_TIME
+        )
+        expect(transaction?.status).toBe(TransactionStatus.SUCCESS)
+    }, TESTS_TIMEOUT)
+})
+
+describe('P1 tests', () => {
+    test(
+        'should generate a key',
+        async () => {
+            const { privateKey, publicKey,  } =
+                await DefaultCrypto.generateSigningKeyPair()
+
+            const serializedKeypair = {
+                privateKey: b64encode(
+                    await DefaultCrypto.serializeSigningKey(privateKey, 'pkcs8'),
+                ),
+                publicKey: b64encode(
+                    await DefaultCrypto.serializeSigningKey(publicKey, 'spki'),
+                ),
+            }
+
+            expect(serializedKeypair.privateKey).not.toBeNull()
+            expect(serializedKeypair.publicKey).not.toBeNull()
+        },
+        TESTS_TIMEOUT,
+    )
+
+    test(
+        'should fetch the current account',
+        async () => {
+            const wallet = await lightsparkClient.getCurrentAccount()
+            expect(wallet?.id).toBeDefined()
+        },
+        TESTS_TIMEOUT,
+    )
+
+    test(
+        'should fetch the current account from unauthorized client',
+        async () => {
+            await expect(unauthorizedLightsparkClient.getCurrentAccount()).rejects.toThrowError()
+        },
+        TESTS_TIMEOUT,
+    )
+
+    test(
+        'should listen current payment requests',
+        async () => {
+            const requests = await lightsparkClient.getRecentPaymentRequests(
+                getRegtestNodeId(),
+                PAGINATION_STEP,
+                BitcoinNetwork.REGTEST,
+            )
+            expect(requests.length).toBe(PAGINATION_STEP)
```

Comment:
```suggestion
            // There will be more transactions than paginated due to repeat testing on this account, so the amount returned should equal PAGINATION_STEP
            expect(requests.length).toBe(PAGINATION_STEP)
```
Let's add this comment so it's clear why transactions will equal PAGINATION_STEP

Reply:
Created a function that will create 20 transactions before this tests

Reply:
We should do less than 20 to save time and prevent flakiness. How about 3?

Reply:
I just reduced the pagination step from 30 to 3. I think this will solve the issue.